checks.d/splunk.py

```diff
@@ -82,6 +85,33 @@ def is_captain(self, instance_tags, url, sessionkey, timeout):
         else:
             return True
 
+    def is_forwarder(self, instance_tags, url, sessionkey, timeout):
+        try:
+            self.get_json(url, '/services/data/inputs/all', instance_tags, sessionkey, timeout)
+        except Exception as inst:
+            return False
+        else:
+            return True
+
+    def do_forwarder_metrics(self, instance_tags, url, sessionkey, timeout):
+        response = self.get_json(url, '/services/admin/inputstatus/TailingProcessor:FileStatus', instance_tags, sessionkey, timeout)
+
+        count = 0
+        for fname in response['entry'][0]['content']['inputs']:
+            input_status = response['entry'][0]['content']['inputs'][fname]
+            # Not every input is actually followed. Those that are have a few
+            # keys. We'll use percent as our signal.
+            if 'percent' in input_status:
+                count += 1
+                status_percent = input_status['percent']
+                self.gauge('splunk.forwarder.read_percentage', input_status['percent'], tags=instance_tags + [
+                    'filename:{0}'.format(fname)
+                ])
+                if status_percent != 100:
+                    self.count('splunk.forwarder.incomplete_files_total', 1, tags=instance_tags)
+
+        self.gauge('splunk.forwarder.files_read_count', count, tags=instance_tags)
```

Comment:
That's what I was suggesting above. Nuking the "this file is fine" gauge and instead *just* reporting those < 100. This would cut down on the metrics chunder.

As you point out we can also report this based on size.